sys/libc/string.c

```diff
@@ -50,4 +52,25 @@ const void *memchk(const void *data, uint8_t c, size_t len)
     return NULL;
 }
 
+int swprintf(string_writer_t *sw, const char *restrict format, ...)
+{
+    va_list args;
```

Comment:
Hm do you think `assert(sw)` is worth it?

Reply:
feels excessive, then we could do that for every function that expects a pointer argument (and why not `assert(format)`?)

Reply:
There's a reason I asked instead of suggesting :p 

Okay, what about the init function? Ensuring the developer actual provides a buffer could save some stupidity.